resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/constants.scala

```
@@ -19,10 +19,12 @@ package org.apache.spark.deploy.kubernetes
 package object constants {
   // Labels
   private[spark] val SPARK_DRIVER_LABEL = "spark-driver"
-  private[spark] val SPARK_APP_ID_LABEL = "spark-app-id"
-  private[spark] val SPARK_APP_NAME_LABEL = "spark-app-name"
+  private[spark] val SPARK_APP_ID_LABEL = "spark-app-selector"
   private[spark] val SPARK_EXECUTOR_ID_LABEL = "spark-exec-id"
   private[spark] val SPARK_ROLE_LABEL = "spark-role"
+  private[spark] val SPARK_POD_DRIVER_ROLE = "driver"
+  private[spark] val SPARK_POD_EXECUTOR_ROLE = "executor"
+  private[spark] val SPARK_APP_NAME_ANNOTATION = "spark-app-name"
```

Should "reserved" identifiers like "spark-app-name" in annotation keys be documented?

---

They could be but the error message might be self-documenting enough, and I'm not sure if it's worth the space. Do you anticipate this being an annotation that people will want to set?

---

It seems more like a low-prob corner case.  If it's a lot of doc, then might not be worth it.  If it's low effort then I'd consider it worth doing

---

Mostly would be worried if there is potential for silent failures

---

We send the notification here: https://github.com/apache-spark-on-k8s/spark/pull/331/files#diff-8c861c0709460ebb9571f0d44791b6beR92